sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/EventTimeWatermarkExec.scala

```diff
@@ -129,3 +129,37 @@ case class EventTimeWatermarkExec(
   override protected def withNewChildInternal(newChild: SparkPlan): EventTimeWatermarkExec =
     copy(child = newChild)
 }
+
+case class UpdateEventTimeColumnExec(
+    eventTime: Attribute,
+    delay: CalendarInterval,
+    child: SparkPlan) extends UnaryExecNode {
+
+  val delayMs = EventTimeWatermark.getDelayMs(delay)
+  override protected def doExecute(): RDD[InternalRow] = {
+    child.execute()
+  }
+
+  // Update the metadata on the eventTime column to include the desired delay.
+  override val output: Seq[Attribute] = child.output.map { a =>
```

Comment:
Same here. We could probably reuse the functions ?

Reply:
Done. Consolidated the Exec and logical operator code duplication as well.